src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -163,32 +177,53 @@ public int GetHashCode(object obj)
 
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public sealed class CultureAwareComparer : StringComparer
+    public sealed class CultureAwareComparer : StringComparer, ISerializable
     {
+        internal const CompareOptions ValidCompareMaskOffFlags = ~(CompareOptions.IgnoreCase | CompareOptions.IgnoreSymbols | CompareOptions.IgnoreNonSpace | CompareOptions.IgnoreWidth | CompareOptions.IgnoreKanaType | CompareOptions.StringSort);
+
         private readonly CompareInfo _compareInfo; // Do not rename (binary serialization)
-        private readonly bool _ignoreCase; // Do not rename (binary serialization)
 
+        [OptionalField]
+        private CompareOptions _options;
+        
         internal CultureAwareComparer(CultureInfo culture, bool ignoreCase)
         {
             _compareInfo = culture.CompareInfo;
-            _ignoreCase = ignoreCase;
+            _options = ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
         }
 
-        private CompareOptions Options => _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        internal CultureAwareComparer(CultureInfo culture, CompareOptions compareOptions)
+        {
+            _compareInfo = culture.CompareInfo;
+            _options = compareOptions;
+        }
+
+        internal CultureAwareComparer(SerializationInfo info, StreamingContext context)
```

Comment:
This should be private as this constructor will only be invoked by the serialization engine via reflection and because the type is sealed